general/networking/nginx.md

```diff
@@ -10,65 +10,85 @@ title: Nginx Reverse Proxy
-#    }
-#    location /socket {
-#        # Proxy Jellyfin Websockets traffic
-#        proxy_pass http://SERVER_IP_ADDRESS:8096/socket;
-#        proxy_http_version 1.1;
-#        proxy_set_header Upgrade $http_upgrade;
-#        proxy_set_header Connection "upgrade";
-#        proxy_set_header Host $host;
-#        proxy_set_header X-Real-IP $remote_addr;
-#        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-#        proxy_set_header X-Forwarded-Proto $scheme;
-#        proxy_set_header X-Forwarded-Protocol $scheme;
-#        proxy_set_header X-Forwarded-Host $http_host;
-#    }
-#}
+    # listen 443 ssl http2;
+    server_name DOMAIN_NAME;
+
+   # use a variable to store the upstream proxy
+   # in this example we are using a hostname which is resolved via DNS
+   # (if you aren't using DNS remove the resolver line and change the variable to point to an IP address e.g `set $jellyfin 127.0.0.1`)
+   set $jellyfin jellyfin;
+   resolver 127.0.0.1 valid=30;
+
+    #ssl_certificate /etc/letsencrypt/live/DOMAIN_NAME/fullchain.pem;
+    #ssl_certificate_key /etc/letsencrypt/live/DOMAIN_NAME/privkey.pem;
+    #include /etc/letsencrypt/options-ssl-nginx.conf;
+    #ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
+    #add_header Strict-Transport-Security "max-age=31536000" always;
+    #ssl_trusted_certificate /etc/letsencrypt/live/DOMAIN_NAME/chain.pem;
+    #ssl_stapling on;
+    #ssl_stapling_verify on;
+
+    # Security / XSS Mitigation Headers
+    add_header X-Frame-Options "SAMEORIGIN";
+    add_header X-XSS-Protection "1; mode=block";
+    add_header X-Content-Type-Options "nosniff";
+
+     # Content Security Policy
+     # See: https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
+     # Enforces https content and restricts JS/CSS to origin
+     # External Javascript (such as cast_sender.js for Chromecast) must be whitelisted.
+     #add_header Content-Security-Policy "default-src https: data: blob:; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' https://www.gstatic.com/cv/js/sender/v1/cast_sender.js; worker-src 'self' blob:; connect-src 'self'; object-src 'none'; frame-ancestors 'self'";
+
+    location / {
+        # Proxy main Jellyfin traffic
+        proxy_pass http://$jellyfin:8096/;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Forwarded-Protocol $scheme;
+        proxy_set_header X-Forwarded-Host $http_host;
+
+        # Disable buffering when the nginx proxy gets very resource heavy upon streaming
+        proxy_buffering off;
+    }
+
+    # location block for /web - This is purely for aesthetics so /web/#!/ works instead of having to go to /web/index.html/#!/
+    location ~ ^/web/$ {
+        # Proxy main Jellyfin traffic
+        proxy_pass http://$jellyfin:8096/web/index.html;
```

Comment:
```suggestion
        proxy_pass http://$jellyfin:8096/web/index.html/;
```

Can you test this fix? On a lot of these, we should be adding terminating slashes.